src/config.ts

```diff
@@ -19,16 +23,54 @@ export interface FunctionConfig {
   zipGo?: boolean
 }
 
+interface FunctionConfigFile {
+  config: FunctionConfig
+  version: number
+}
+
 type GlobPattern = string
 
-export type Config = Record<GlobPattern, FunctionConfig>
+type Config = Record<GlobPattern, FunctionConfig>
+type FunctionWithoutConfig = Omit<FunctionSource, 'config'>
+
+const getConfigForFunction = async ({
+  config,
+  configFileDirectories,
+  func,
+  featureFlags,
+}: {
+  config?: Config
+  configFileDirectories?: string[]
+  func: FunctionWithoutConfig
+  featureFlags: FeatureFlags
+}): Promise<FunctionConfig> => {
+  const fromConfig = getFromMainConfig({ config, func })
+
+  // We try to read from a function config file if the function directory is
+  // inside one of `configFileDirectories`.
+  const shouldReadConfigFile =
+    featureFlags.project_deploy_configuration_api_use_per_function_configuration_files &&
+    configFileDirectories?.some((directory) => func.srcDir.startsWith(directory))
```

Comment:
Is there the potential for the `startsWith` to catch false positives? For example:

- `configFileDirectories`: [`"foo/bar"`]
- `func.srcDir`: `"foo/bar-and-something-else"`

Reply:
I changed the logic and use the package `is-path-inside` which ensures this works correctly. Also adjusted the test to cover for this. See new commits.